### PR TITLE
[#9780] Fix JS error in IE10

### PR DESCRIPTION
**Pull request by msamsel**
Fix which generated throwing an error in IE10 and below.
In case of IE `range.parentElement()` was undefined and there was reference to `ownerDocument` which throws an error.

Unit test wasn't possible to make. Problem was with reproducing selection in a similar way as IE has. There was necessary to make selection with `type` other than `None` to have test which check second condition in `if` statement. Selection made by regular cursor had `Text` type, so error was visible. Bender tools create selection with `None` type only :(.

### CODE REVIEW

File: core/selection.js [hunk 1]
```diff
@@ -1168,7 +1168,7 @@
 					range = sel.createRange();
 
 				// The selection range is reported on host, but actually it should applies to the content doc.
-				if ( sel.type != 'None' && range.parentElement().ownerDocument == doc.$ )
+				if ( sel.type != 'None' && ( range.parentElement() || {}).ownerDocument == doc.$ )
```
Comment (f1ames) [inline core/selection.js]: We could also check if `range.parentElement()` exists, not sure which one is more readable. Please remember about proper formatting like `( range.parentElement() || {} )` (missing space before closing parenthesis).

Comment (msamsel) [inline core/selection.js]: Ok so i change it also a little bit:
```
if ( sel.type != 'None' && range.parentElement() && range.parentElement().ownerDocument == doc.$ )
```
So if 2nd part of condition failed then 3rd one won't be evaluated.

**Comment by msamsel** [comment]
Manual tests was moved to `tickets`
I added manual tests for 16820.
Also modified a little bit the condition to have more adequate syntax in this case.
Test was restricted to IE only in version below 11.

File: core/selection.js [hunk 2]
```diff
@@ -1168,7 +1168,8 @@
 					range = sel.createRange();
 
 				// The selection range is reported on host, but actually it should applies to the content doc.
-				if ( sel.type != 'None' && range.parentElement().ownerDocument == doc.$ )
+				// #9780 added check if range.parentElement() exists
```
Comment (f1ames) [inline core/selection.js]: Remember to put ticket number at the end of the comment like:
> The `parentElement` may be null for read only mode in IE10 and below. (#9780)

Also try to describe the case why this change was introduced (like above).

File: tests/tickets/16820/manual/ie10hrline.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<div id="editor">
```
Comment (f1ames) [inline tests/tickets/16820/manual/ie10hrline.html]: Since this tests are already in dedicated folder `tests/16820/` there is no need for such specific naming. You may simply name files `1.html`, `1.md` (unless the issue covers more cases which should be clearly distinguishable).

File: tests/tickets/16820/manual/ie10hrline.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+@bender-tags: selection, tc, 4.7.1, 16820
+@bender-ckeditor-plugins: wysiwygarea,toolbar,horizontalrule,newpage
```
Comment (f1ames) [inline tests/tickets/16820/manual/ie10hrline.md]: I think `newpage` dependency is not needed. You may just initialize the test with empty editor which will also simplify the test to 2 steps :)

File: tests/tickets/9780/manual/ie10readonly.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<div id="editor"></div>
```
Comment (f1ames) [inline tests/tickets/9780/manual/ie10readonly.html]: I would also update this test name to `1.html` and `1.md` as mentioned in another comment.

File: tests/tickets/16820/manual/ie10hrline.html [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+	<p>Foo</p>
+</div>
```
Comment (f1ames) [inline tests/tickets/16820/manual/ie10hrline.html]: I think the more readable formatting will be (newline separating editor html and js "logic"):

```
<div id="editor">
	<p>Foo</p>
</div>

<script>
	if ( !( CKEDITOR.env.ie && CKEDITOR.env.version < 11 ) ) {
        bender.ignore();
    }

	CKEDITOR.replace('editor', {
		height: 300,
		enterMode: CKEDITOR.ENTER_BR
	} );
</script>
```
